components/motor/gpio/motor_encoder.go

```diff
@@ -102,26 +102,40 @@ func newEncodedMotor(
 	em.encoder = realEncoder
 
 	// setup control loop
-	if len(motorConfig.ControlLoop.Blocks) != 0 {
-		em.controlLoopConfig = em.createControlLoopConfig()
-		cLoop, err := control.NewLoop(em.logger, em.cfg.ControlLoop, em)
+	if motorConfig.ControlParameters != nil {
+		// create control loop config with PID values from config
+		em.controlLoopConfig = em.createControlLoopConfig(
+			motorConfig.ControlParameters.P,
+			motorConfig.ControlParameters.I,
+			motorConfig.ControlParameters.D,
+		)
+
+		cLoop, err := control.NewLoop(em.logger, em.controlLoopConfig, em)
 		if err != nil {
 			em.logger.Error(err)
 		}
 		if err = cLoop.Start(); err != nil {
 			em.logger.Error(err)
 		}
 		em.loop = cLoop
+
+		// validate control loop config
 		if err = em.validateControlConfig(cancelCtx); err != nil {
 			return nil, err
 		}
-		pidBlock, err := em.loop.ConfigAt(cancelCtx, "PID")
-		if err != nil {
-			return nil, err
-		}
-		if pidBlock.Attribute["kP"].(float64) == 0.0 && pidBlock.Attribute["kI"].(float64) == 0.0 && pidBlock.Attribute["kD"].(float64) == 0.0 {
-			em.loop.SetTuning(cancelCtx, true)
+
+		// SetTuning to true if all PID values are 0
+		if motorConfig.ControlParameters.P == 0.0 &&
+			motorConfig.ControlParameters.I == 0.0 &&
+			motorConfig.ControlParameters.D == 0.0 {
+			em.loop.SetTuning(context.Background(), true)
```

Comment:
Is this background context canceled by anything in the loop? Or will it continue returning nil and blocking on a done call in the controls package?

Reply:
I don't think it's canceled by anything... I'm not sure it's necessary to even have a context for this function

Reply:
this function should be removed in a future pr